### DISCUSSION TITLE
Qt5 may not suit for GUI of new release  groops (on Ubuntu20.04)?

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#209] [Q&A]
Hello developers,
I'm trying to deploy the new release version of groops on ubuntu 20.04. But the GUI can not be successfully built according to the installation guide of Linux-ubuntu. So I guess Qt5 may not be compatible with the newly version? Because deploying higher version QT is inconvenient  on linux OS, something like .exe for GUI installation on ubuntu or updating the installation-guide may be helpful. Looking forward to your reply.
regards,
Fan

**Comment by tmayerguerr** [comment]
Hi Fan,

Hmm, this should work with Qt5. Can you give me the error message and which QT version you are using (with  `qmake -v`)?

regards
Torsten

**Reply by FFFFFG-IGG** [reply]
Hello,

This is screenshot of qmake -v:
![image](https://github.com/groops-devs/groops/assets/59271400/e0dcac68-add4-4880-9558-8e1c17d4ee0e)
and the error of building GUI  with QT  are:
![image](https://github.com/groops-devs/groops/assets/59271400/bc46f2f9-9a62-4d61-a3cb-d43b06014aaa)

looking forward to you reply.

regards,
Fan

**Reply by FFFFFG-IGG** [reply]
Note that QT is install with command " sudo apt-get install qtbase5-dev ", here is the screenshot after installation:
![image](https://github.com/groops-devs/groops/assets/59271400/4a40a4f5-ee42-43cc-b2b9-3b9c1d8fd1ee)

**Comment by zhedumi** [comment]
hi,
Its curious that QT seems to be versions behind in Ubuntu in comparison to Debian. Or is the repository not updated? Might want to try sudo apt update and then sudo apt install qtbase5-dev again. 
Cause with the QT 5.14 and higher it should be working as intendent
![image](https://github.com/groops-devs/groops/assets/77068736/d3015bbd-f18f-4abe-95d9-eb25146f60ce) .

Cause Qt:SkipEmptyParts (And has replaced the older QString::SkipEmptyParts) is the correct now in QT 5.14 and QT 6.XX and higher according to
https://doc.qt.io/qt-5/qstring-obsolete.html#SplitBehavior-enum
https://doc.qt.io/qt-5/qt.html#SplitBehaviorFlags-enum

Therefore it is safe to say that this is a QT Linux problem and not a GROOPS issue

**Comment by tmayerguerr** [comment]
Hi,
The easiest way is to replace `Qt::SkipEmptyParts` by `QString::SkipEmptyParts`  in the source. Then it should also work with the old Qt version.
Torsten
